### ISSUE TITLE
Trap focus

### ISSUE THREAD

**Issue by DHedgecock** [crystal-ball/componentry#98]
When users enter elements like a modal or dropdown, focus should be trapped in that element for accessibility. Tabbing through focuseable elements should return to the first available element in the element similar to how Bootstrap handles modals: https://github.com/twbs/bootstrap/blob/v4-dev/js/src/modal.js#L275
